p2p/simulations/network.go

```diff
@@ -83,11 +84,15 @@ func (net *Network) Events() *event.Feed {
 func (net *Network) NewNodeWithConfig(conf *adapters.NodeConfig) (*Node, error) {
 	net.lock.Lock()
 	defer net.lock.Unlock()
-
 	if conf.Reachable == nil {
 		conf.Reachable = func(otherID enode.ID) bool {
+			log.Trace("kademlia.Reachable", "thisNode", conf.ID.TerminalString(), "thatNode", otherID.TerminalString())
 			_, err := net.InitConn(conf.ID, otherID)
-			if err != nil && bytes.Compare(conf.ID.Bytes(), otherID.Bytes()) < 0 {
+			if err != nil {
+				log.Trace("net.InitConn returned an error", "err", err)
+				r := rand.New(rand.NewSource(time.Now().UnixNano()))
+				t := time.Duration(time.Duration(r.Intn(200)) * time.Millisecond)
```

Comment:
I read this as skewing of time offsets in case nodes try to connect simultaneously? Is this actually a solution to the problem?